### ISSUE TITLE
Periods in base filename interpreted as extensions

### ISSUE THREAD

**Issue by iceboundflame** [webdataset/webdataset#237]
It appears that periods in the base part of the filename cause issues. For example the filename `./235342 Track 2.0 (Clean Version).mp3` leads to an unexpected key of `0 (clean version).mp3`. This caused sporadic downstream issues like:

```ValueError: didn't find ['mp3'] in ['__key__', '__url__', '0 (clean version).json', '0 (clean version).mp3']```

Looking into the code, it appears that this is by design in order to support multiple extensions like ".seg.jpg", but it would be nice to mention this in the documentation to avoid surprises.

**Comment by tmbdev** [comment]
Yes, this is by design and pretty much unavoidable; we need some character for multiple extensions, and "." is a natural and common choice. I'll try to add something to the documentation (which, unfortunately, is pretty minimalist to begin with).

WebDataset doesn't impose a lot of restrictions on tar files, but this is one of them; having such conventions makes writing downstream pipelines much easier.

Note that you could fix this in the input pipeline by adding a function mapping the file names between the tariterator and the tarfile_to_samples stages in the pipeline; however, this really isn't recommended. It's better just to deal with this during dataset creation.

Note that in several places, you can use "glob" patterns like "*.mp3" to match extensions.
